numpy/core/shape_base.py

```diff
@@ -177,6 +177,71 @@ def atleast_3d(*arys):
         return res
 
 
+def atleast_nd(n, *arys):
+    """
+    View inputs as arrays with at least `n` dimensions.
+
+    Parameters
+    ----------
+    n : scalar
```

Comment:
I disagree. I think it is important to have `n` as the first parameter to `atleast_nd` from an intuitive point of view. That parameter is required and does not make nearly as much sense at the end of the list.

The changes in `ma` are not very drastic and don't affect the existing cases. I basically added the option to have a "prefix" of non-array args to a class that is private to begin with. All the existing tests should run fine now that I have fixed the syntax for pre-3.5 Python.